### DISCUSSION TITLE
Battery voltage reading

### DISCUSSION THREAD

**Discussion by Nilon123456789** [rbaron/b-parasite#137] [Q&A] [answered]
I am wondering how the battery voltage reading works, since it's using adc AIN2 that is not connected.

**Comment by rbaron** [comment, marked as answer]
Hey @Nilon123456789,

You can see [here](https://github.com/rbaron/b-parasite/blob/426195184d2433f06f4b30ea665348985cdaffb3/code/nrf-connect/prstlib/boards/arm/bparasite_nrf52840/bparasite_nrf52840.dts#LL131C28-L131C41) that we use channel 2 for battery measurement, and it's configured to use the VDD pin as input (via the `NRF_SAADC_VDD` value). Channel 2 refers to the SAADC channel, and it's a different thing from the `AIN2` pin (which is not used for this). You can refer to section 6.23 on [the datasheet](https://infocenter.nordicsemi.com/pdf/nRF52840_PS_v1.1.pdf) for this:

<img width="666" alt="Screenshot 2023-06-09 at 22 34 06" src="https://github.com/rbaron/b-parasite/assets/1573409/c7bdb0c7-4bc3-4e7d-a870-ab13d3fcd100">

**Reply by Nilon123456789** [reply]
Alright thanks

**Reply by Nilon123456789** [reply]
A question just popped in my head, why do you define in the zephyr,user the io-channel adc 2, if your not using it. Is it just to be able to fetch the config of channel 2?

**Reply by rbaron** [reply]
I'm not sure I follow. We do use it [in adc.c:37](https://github.com/rbaron/b-parasite/blob/426195184d2433f06f4b30ea665348985cdaffb3/code/nrf-connect/prstlib/src/adc.c#L37). This is the same strategy used in [this Zephyr sample](https://docs.zephyrproject.org/latest/samples/drivers/adc/README.html).
